app/src/main/java/ch/epfl/sdp/peakar/points/ComputePOIPoints.java

```diff
@@ -37,17 +40,51 @@
      * @param context Context of activity
      */
     public ComputePOIPoints(Context context){
+        ctx = context;
         POIPoints = new ArrayList<>();
         userPoint = UserPoint.getInstance(context);
         userPoint.update();
         getPOIs(userPoint);
     }
 
+
     /**
-     * Gets the POIs for the userPoint
-     * @param userPoint location of the user
+     * Retrieves list of surrounding POIs either from cache
+     * or from provider. If the cached data corresponds to
+     * the user location, no download is made and the POIs
+     * are retrieved from cached file
+     * @param userPoint user location
      */
     private static void getPOIs(UserPoint userPoint){
+        //Retrieve cache instance
+        POICache poiCache = POICache.getInstance();
+        //Check if the user is in the cached BB. If yes
+        //load data from cache / if not download data
+        if(poiCache.isCacheFilePresent(ctx) && poiCache.isUserInBoundingBox(userPoint, ctx))
+            getPOIsFromCache(userPoint);
+        else
+            getPOIsFromProvider(userPoint);
+    }
+
+    /**
+     * Get surrounding POIs from cache
+     * @param userPoint location of the user
+     */
+    private static void getPOIsFromCache(UserPoint userPoint){
+        ArrayList<POIPoint> cachedPOIs = POICache.getInstance().getCachedPOIPoints(ctx);
+        POIPoints.addAll(cachedPOIs.stream().peek(poiPoint ->
+        {
+            poiPoint.setHorizontalBearing(userPoint);
+            poiPoint.setVerticalBearing(userPoint);
+        }).collect(Collectors.toList()));
+        //getLabeledPOIs(userPoint);
```

Comment:
```suggestion
```

if it is something for the future add TODO instead

Reply:
Yes indeed that is something for a later task, I will add a todo instead of removing the comment completely